apps/mobile/src/components/Feed/Socialize/ProcessedText.tsx

```diff
@@ -0,0 +1,255 @@
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType | null>(null);
+  const handleLinkPress = useCallback(() => {
+    bottomSheetRef.current?.present();
+  }, []);
+
+  return (
+    <>
+      <Text className="text-shadow" onPress={handleLinkPress}>
+        {value ?? url}
+      </Text>
+      <WarningLinkBottomSheet redirectUrl={url} ref={bottomSheetRef} />
+    </>
+  );
+};
+
+type MentionProps = {
+  mention: string;
+  mentionRef: ProcessedTextMentionFragment$key;
+};
+
+const MentionComponent = ({ mention, mentionRef }: MentionProps) => {
+  const mentionData = useFragment(
+    graphql`
+      fragment ProcessedTextMentionFragment on Mention {
+        __typename
+        entity {
+          __typename
+          ... on GalleryUser {
+            __typename
+            username
+          }
+          ... on Community {
+            __typename
+            contractAddress {
+              __typename
+              address
+              chain
+            }
+          }
+        }
+      }
+    `,
+    mentionRef
+  );
+
+  const navigation = useNavigation<MainTabStackNavigatorProp>();
+
+  const onMentionPress = useCallback(() => {
+    if (mentionData.entity?.__typename === 'GalleryUser') {
+      navigation.navigate('Profile', {
+        username: mentionData.entity.username ?? '',
+      });
+      return;
+    }
+
+    if (mentionData.entity?.__typename === 'Community') {
+      navigation.navigate('Community', {
+        contractAddress: mentionData.entity.contractAddress?.address ?? '',
+        chain: mentionData.entity.contractAddress?.chain ?? '',
+      });
+    }
+  }, [mentionData, navigation]);
+
+  return (
+    <Text className="text-shadow" onPress={onMentionPress}>
+      {mention}
+    </Text>
+  );
+};
+
+type CommentProps = {
+  text: string;
+  mentionsRef?: ProcessedTextFragment$key;
+} & TextProps;
+
+type CommentElement = {
+  type: 'mention' | 'url' | 'markdown-link';
+  value: string;
+  start: number;
+  end: number;
+  mentionRef?: ProcessedTextMentionFragment$key;
+  url?: string;
+};
+
+// Makes a raw text value display-ready by converting urls to link components
+export default function ProcessedText({ text, mentionsRef = [], ...props }: CommentProps) {
+  const mentions = useFragment(
+    graphql`
+      fragment ProcessedTextFragment on Mention @relay(plural: true) {
+        __typename
+        interval {
+          __typename
+          start
+          length
+        }
+        entity {
+          __typename
+          ... on GalleryUser {
+            __typename
+            username
+          }
+          ... on Community {
+            __typename
+            contractAddress {
+              __typename
+              address
+              chain
+            }
+          }
+        }
+        ...ProcessedTextMentionFragment
+      }
+    `,
+    mentionsRef
+  );
+
+  const processedText = useMemo(() => {
+    const elements: CommentElement[] = [];
```

Comment:
I think decomposing this into different helper functions could help make this easier to read. it could also make debugging the issue from slack a bit easier